api/app/serializers/spree/v2/storefront/product_serializer.rb

```diff
@@ -54,16 +54,22 @@ class ProductSerializer < BaseSerializer
 
         # all images from all variants
         has_many :images,
-          object_method_name: :variant_images,
-          id_method_name: :variant_image_ids,
-          record_type: :image,
-          serializer: :image
-
-        has_one  :default_variant,
-          object_method_name: :default_variant,
-          id_method_name: :default_variant_id,
-          record_type: :variant,
-          serializer: :variant
+                 object_method_name: :variant_images,
```

Comment:
@szymoniwacz let's also do the same for platform serializer, please - https://github.com/spree/spree/blob/master/api/app/serializers/spree/api/v2/platform/product_serializer.rb#L47